### PR TITLE
Realtime kit

### PR DESCRIPTION
**Pull request by varshini2402**

### CODE REVIEW

**Comment by kevmo314** [comment]
@juaoose can you review this PR?

File: remote/lib/main.dart [hunk 1]
```diff
@@ -1,115 +1,193 @@
...
 
-void main() {
-  runApp(const MyApp());
-}
-
+void main() => runApp(MyApp());
+  
 class MyApp extends StatelessWidget {
-  const MyApp({Key? key}) : super(key: key);
+  var myMenuItems = <String>[
```
Comment (juaoose) [inline remote/lib/main.dart:7]: Shouldn't this be final? Considering its a stateless widget.

**Review by juaoose — CHANGES_REQUESTED** [review]
There are also some conventions that you could check by running `flutter analyze`, most of them are very minor.

**Review by juaoose — APPROVED** [review]
As we discussed, try to work on the conventions and analyzer warnings on the next PRs :)

File: remote/lib/alert.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
+
+
+
+import 'package:flutter/material.dart';
+
+class Class3 extends StatelessWidget {
```
Comment (juaoose) [inline remote/lib/alert.dart:6]: Probably better to also name this class with something more meaningful.

File: remote/lib/main.dart [hunk 3]
```diff
@@ -1,115 +1,498 @@
...
+                },),
+
+            ],
+          ),
+        )
+        
+       );
+}
+/*
+void main() => runApp(MyApp());
```
Comment (juaoose) [inline remote/lib/main.dart:307]: Could we remove this commented code?

File: functions/src/index.ts [hunk 4]
```diff
@@ -1,102 +1,120 @@
...
+      res.status(400).send();
+      return;
+    }
+    const doc = await admin
+      .firestore()
+      .collection("channels")
+      .doc(channelId)
+      .get();
+    res.status(200).send(doc);
+    if(doc === undefined){
```
Comment (juaoose) [inline functions/src/index.ts:26]: I think for the document you would need to do `!doc.exists`, the undefined assignment I believe is the bevahior for `document.get("fieldName")`.

File: functions/src/index.ts [hunk 5]
```diff
@@ -1,102 +1,120 @@
...
+    if (!validate(channelId)) {
+      res.status(400).send();
+      return;
+    }
+    const doc = await admin
+      .firestore()
+      .collection("channels")
+      .doc(channelId)
+      .get();
+    res.status(200).send(doc);
```
Comment (juaoose) [inline functions/src/index.ts:25]: This should be executed after you check that the document exists and has data, and its probably better to only return the data of the doc: `res.status(200).send(doc.data());`

**Review by juaoose — CHANGES_REQUESTED** [review]

**Comment by kevmo314** [comment]
Can the function changes also be pulled out into a separate pull request?

File: remote/lib/settings.dart [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+           child:IconButton(onPressed:  (){ 
+          themeNotifier.isDark
+          ? themeNotifier.isDark=false:themeNotifier.isDark=true;
+        }, icon: Icon(themeNotifier.isDark?Icons.wb_sunny:Icons.nightlight_round)))
+      ],
+    ),
+
+            //buildAccountOptionRow(context, "Change theme"),
+            buildNotificationOptionRow("Wake lock", true),
+            buildNotificationOptionRow("Confirm before streaming", true),
```
Comment (juaoose) [inline remote/lib/settings.dart:71]: I guess we could remove these items we don't use, but we can do that in another PR.

File: remote/lib/operations.dart [hunk 7]
```diff
@@ -0,0 +1,1016 @@
...
+        context: context,
+        builder: (context) {
+          return AlertDialog(
+            title: const Text('Confirm before Streaming'),
+            content: const Text("Are you sure you want to start streaming"),
+           
+            actions: <Widget>[
+              ElevatedButton(
+                child:  const Text('Yes'),
+                onPressed: () async {
```
Comment (juaoose) [inline remote/lib/operations.dart:563]: The dialog should also be dismissed when "Yes" is selected.

File: remote/lib/alert.dart [hunk 8]
```diff
@@ -0,0 +1,72 @@
+
+
+
+import 'package:flutter/material.dart';
+
+class Alert extends StatelessWidget {
```
Comment (juaoose) [inline remote/lib/alert.dart:6]: I believe this class is not being used, we can remove it.

**Review by juaoose — APPROVED** [review]
Left some commets we can address in a new PR since this is one is too big already.

Also, I think you could simplify the "uuid" and "url" management, I think we don't really need to store the full URL when read from the QR code and just assume "https://kit.rtirl.com/api/" is our base. This way we could always just store an UUID and remove some code from the operations file.

**Pull request merged**